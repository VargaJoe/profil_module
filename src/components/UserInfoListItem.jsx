import React, { Component } from 'react';
import Timestamp 			from 'react-timestamp';

class UserInfoListItem extends Component {
	constructor(props) {
		super(props);
		this.state = {
			name : "List"
		};
	}

	render () {
		let userValue;
		switch (this.props.infoType) {
			case 0:
				userValue = this.props.value;
				break;
			
			case 1:
				userValue = (
					<a href="/">
						{this.props.value}
					</a>
				);
				break;

			case 2:
				userValue = <Timestamp time={this.props.value} utc={false} format='date' precision={2}/>;
				break;
		
			default:
				break;
		}
		// if (this.props.infoType) {
		// 	userValue = (
		// 		<a href="/">
		// 			{this.props.value}
		// 		</a>
		// 	);
		// } else {
		// 	userValue = this.props.value;
		// }

		return (
			<div className="user__global_info__list__item">
				<div>
					{this.props.name} :
				</div>
				<div>
					{userValue}
				</div>
			</div>
		)
	}
};

export default UserInfoListItem;
