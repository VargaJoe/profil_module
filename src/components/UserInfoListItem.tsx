import * as React				from 'react';
// import * as Timestamp 			from 'react-timestamp';

class UserInfoListItem extends React.Component<any, any> {

	render () {
		let userValue;
		switch (this.props.infoType) {
			case 0:
				userValue = this.props.value;
				break;
			
			case 1:
				userValue = (
					<a href="/">
						{this.props.value}
					</a>
				);
				break;

			case 2:
				userValue = (
					<select>
						<option selected={true} value={this.props.value}>{this.props.value}</option> 
					</select>
				);
				break;
		
			default:
				break;
		}

		return (
			<div className="user__global_info__list__item">
				<div className="user__global_info__list__item--name">
					{this.props.name} :
				</div>
				<div className="user__global_info__list__item--value">
					{userValue}
				</div>
			</div>
		);
	}
}

export default UserInfoListItem;
