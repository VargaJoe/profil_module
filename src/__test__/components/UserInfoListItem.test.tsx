import * as React 						  	from 'react';
import UserInfoListItem 					from '../../components/UserInfoListItem';
import { 
		configure, 
		shallow,
		mount,
		render } 							from 'enzyme';
import * as Adapter 					  	from 'enzyme-adapter-react-16';
import toJson 				      			from 'enzyme-to-json';

configure( {adapter: new Adapter()} );

describe('<UserInfoListItem /> shallow rendering', () => {

	const props = {
		name: 'Test item',
		value: 'Item 1',
		infoType: 0
	};
		
	const userinfolist = mount(<UserInfoListItem {...props}/>); 
		
	// test Snapshot 
	it('Match to snapshot', () => {
		expect(toJson(userinfolist)).toMatchSnapshot();
	});

	it('Get a simple text value', () => {
		expect(userinfolist.find('.user__global_info__list__item--name').text()).toBe('Test item :');
		expect(userinfolist.find('.user__global_info__list__item--value').text()).toBe('Item 1');
	});

	it('Get a link value', () => {
		const props2 = {
			name: 'Test item',
			value: 'Item 1',
			infoType: 1
		};
		const userinfolist2 = render(<UserInfoListItem {...props2}/>); 
		expect(userinfolist2.find('.user__global_info__list__item--name').text()).toBe('Test item :');
		expect(userinfolist2.find('.user__global_info__list__item--value a').text()).toBe('Item 1');
	}); 

	it('Get a select box value', () => {
		const props3 = {
			name: 'Test item',
			value: 'Item 1',
			infoType: 2
		};
		const userinfolist3 = render(<UserInfoListItem {...props3}/>); 
		expect(userinfolist3.find('.user__global_info__list__item--name').text()).toBe('Test item :');
		expect(userinfolist3.find('.user__global_info__list__item--value select').val()).toBe('Item 1');
	});
		
	it('Not added value type', () => {
		const props4 = {
			name: 'Test item',
		};
		const userinfolist4 = render(<UserInfoListItem {...props4}/>); 
		expect(userinfolist4.find('.user__global_info__list__item--name').text()).toBe('Test item :');
		expect(userinfolist4.find('.user__global_info__list__item--value').text()).toBe('');
	});
});
 