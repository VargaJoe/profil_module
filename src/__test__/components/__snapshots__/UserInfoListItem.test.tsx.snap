// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfoListItem /> shallow rendering Match to snapshot 1`] = `
<div
  className="user__global_info__list__item"
>
  <div
    className="user__global_info__list__item--name"
  >
    Test item
     :
  </div>
  <div
    className="user__global_info__list__item--value"
  >
    Item 1
  </div>
</div>
`;
